frontend/front/src/pages/Public/Components/CanonicalizationLoader.js

```diff
@@ -0,0 +1,60 @@
+import React, { forwardRef, useEffect, useMemo, useState } from "react";
+import { useGetHomeQuery } from "../../../api/apiSlice";
+import Loader from "../../../components/Loader";
+import { Box } from "@mui/material";
+
+const CanonicalizationLoader = forwardRef(({ homeId, onResolved }, ref) => {
+  const [stopPolling, setStopPolling] = useState(false);
+  const {
+    data: validationData,
+    isLoading: isLoadingValidation,
+    isError: validationError,
+  } = useGetHomeQuery(homeId, { pollingInterval: stopPolling ? 0 : 300 });
+
+  const isUncanonicalized = useMemo(
+    () => validationData?.status === "uncanonicalized",
+    [validationData]
+  );
+
+  const isCanonicalized = useMemo(
+    () => validationData?.status === "canonicalized",
+    [validationData]
+  );
+
+  const isUnrecognized = useMemo(
+    () => validationData?.status === "unrecognized",
+    [validationData]
+  );
+
+  useEffect(() => {
+    if (isLoadingValidation) return;
```

Comment:
I thought best practice was to always have the {} here, but if prettier and eslint aren't complaining then whatever

Reply:
I'm a fan of implicit returns for one-line booleans like this.